File: src/p_3_creational_patterns/p_10_prototype/CompanyAddress.java
package p_5_creational_patterns.p_10_prototype;

public class CompanyAddress extends AddressPrototype{

    private String title;
    private String email;

    public CompanyAddress(String title, String email, String city, String address, String no) {
        this.title = title;
        this.email = email;
        setCity(city);
        setAddress(address);
        setNo(no);
    }

    public String getTitle() {
        return title;
    }

    public void setTitle(String title) {
        this.title = title;
    }

    public String getEmail() {
        return email;
    }

    public void setEmail(String email) {
        this.email = email;
    }

}

File: Net/03_creational_patterns/06_prototype_pattern/CompanyAddress.cs
namespace _06_prototype_pattern
{
    public class CompanyAddress : AddressPrototype
    {
        private string workNum;

        public CompanyAddress(string workNum, string city)
        {
            setCity(city);
            this.workNum = workNum;
        }

        public string getWorkNum()
        {
            return workNum;
        }

        public void setWorkNum(string workNum)
        {
            this.workNum = workNum;
        }
    }
}

